lib/logdna/writer.go

```diff
@@ -129,8 +131,12 @@ func extractToken(u *url.URL) (token string, err error) {
 
 func extractTags(u *url.URL, q url.Values) string {
 	tags := q["tag"]
-	list := make([]string, 0, len(tags))
+	if len(tags) == 0 {
+		tags = "";
+	}
+
+	list := make([]string, 0)
 	list = append(list, tags...)
 
-	return strings.Join(list, ",")
+	return fmt.Sprintf("tags=\"%s\"", strings.Join(list, ","))
```

Comment:
All good,

If there's no `tag` in the query string it'll return the zero-value of a `[]string` with is an empty slice, so `strings.Join` will return an empty string and the function returns `tags=""`.

I believe this matches the behavior you're expecting.